package/src/hydra-ai/hydra-ai-backend.ts

```diff
@@ -1,23 +1,52 @@
 import "server-only";
 import AIService from "./ai-service";
 import { ComponentChoice } from "./model/component-choice";
-import { ComponentMetadata } from "./model/component-metadata";
+import { AvailableComponents, ComponentMetadata } from "./model/component-metadata";
 import { InputContext } from "./model/input-context";
+import { drizzle, NodePgDatabase } from "drizzle-orm/node-postgres";
+import { Pool } from "pg";
+import { registeredComponents } from "../db/schema";
+import { ComponentPropsMetadata } from "./model/component-props-metadata";
 
 export default class HydraBackend {
   private aiService: AIService;
+  private dbConnection: NodePgDatabase;
 
-  constructor(
-    openAIKey: string,
-    openAIModel = "gpt-4o",
-    systemInstructions?: string
-  ) {
-    this.aiService = new AIService(openAIKey, openAIModel, systemInstructions);
+  constructor(dbConnectionUrl: string, openAIKey: string, openAIModel = "gpt-4o") {
+    const pool = new Pool({
+      connectionString: dbConnectionUrl,
+    });
+  
+    this.dbConnection = drizzle(pool);
+    this.aiService = new AIService(openAIKey, openAIModel);
+  }
+
+  public async registerComponent(
+    name: string,
+    description: string,
+    propsDefinition?: ComponentPropsMetadata
+  ): Promise<boolean> {
+    try {
+      let _ = await this.dbConnection.insert(registeredComponents).values({
```

Comment:
nit: Instead of `let _` we can just do `await this.db...` without assigning a variable